### DISCUSSION TITLE
pipenv not setting VIRTUAL_ENV

### DISCUSSION THREAD

**Discussion by MrPaulAR** [JanDeDobbeleer/oh-my-posh#5527] [Troubleshoot]
I'm moving from powerlevel10k to oh-my-posh and working through the process of building my prompt.

The only oddity I'm having is that the python section doesn't seem to be working.  I've narrowed this down to when I execute `pipenv shell` the variable VIRTUAL_ENV isn't being set.  If I disable oh-my-posh and fall back to powerlevel10k the variable exists.

Below is the oh-my-posh config.  It's worth noting that I do have *.py files in the folder and have tried setting display_mode as defined in https://github.com/JanDeDobbeleer/oh-my-posh/issues/408 but no joy.

```json
                {
                    "type": "python",
                    "foreground": "#00afaf",
                    "template": "  {{ if .Error }}{{ .Error }}{{ else }}{{ if .Venv }}{{ .Venv }} {{ end }}{{ .Full }}{{ end }} "
                }
```

Below is the output when I start the `pipenv shell`.
![image](https://github.com/user-attachments/assets/c9828b65-a594-4b08-af54-005a82bebb18)

When I edit the .zshrc and rollback to p10k this is the result.
![image](https://github.com/user-attachments/assets/c6070325-5e2f-4c79-a82b-a906336fdebf)

**Comment by MrPaulAR** [comment]
This issue seems to be related to Arch.  Created a new user account and did the following steps in Arch.

* Installed `asdf-vm` and `oh-my-posh`
* Installed python via asdf
  * `asdf plugin-add python`
  * `asdf install python latest`
  * `asdf global python latest`
* Installed pipenv
  * `pip install pipenv`
* Created an launched a shell via pipenv
  * `mkdir temprepo`
  * `cd temprepo`
  * `pipenv shell`

As you can see from the below this works.

```bash
arch-lxc% cd temprepo 
arch-lxc% pipenv shell
Launching subshell in virtual environment...
arch-lxc%  . /home/tmpuser/.local/share/virtualenvs/temprepo-yT8lAnC9/bin/activate
(temprepo) arch-lxc% 
(temprepo) arch-lxc% echo $VIRTUAL_ENV
/home/tmpuser/.local/share/virtualenvs/temprepo-yT8lAnC9
```

Now enable agnoster theme in oh-my-posh as I'm positive it has the python section enabled.  Added the line below to .zshrc and logged out/in.

```
arch-lxc% eval "$(oh-my-posh init zsh --config 'https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/powerlevel10k.omp.json')"
```

As the screenshot inticates this didn't work.

![image](https://github.com/user-attachments/assets/f86a29b2-faff-4fe5-8fa0-c753d5803f90)

At this point I pivoted to a Ubuntu system and did the same steps but proceeded straight to agnoster in oh-my-posh.  This works.
![image](https://github.com/user-attachments/assets/6519843b-5a4c-4765-8a2b-61c27cfa3f18)


At the end of this the .zshrc file for both the tempuser account on Arch and Ubuntu were exactly the same.

```
export PATH=$PATH:$HOME/.local/bin

. "$HOME/.asdf/asdf.sh"
fpath=(${ASDF_DIR}/completions $fpath)
autoload -Uz compinit && compinit

eval "$(oh-my-posh init zsh --config 'https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/agnoster.omp.json')"
```

**Reply by CVirus** [reply]
I'm seeing the same issue on Mac.
I suspect your Ubuntu has an older version of oh-my-posh compared to arch?
I think a recent oh-my-posh version introduced this bug, because it was working for me some time ago.

**Reply by JanDeDobbeleer** [reply]
@CVirus read my reply. My working theory is that the activate script fails because it tries to override the prompt and fails due to the options we set (which we are allowed to).

**Comment by CVirus** [comment]
I ran a couple of tests and it seems that the issue was introduced in oh-my-posh v23.8.0

**Reply by MrPaulAR** [reply]
Thanks for confirming it's not just me and possibly identifying the root cause.

**Reply by CVirus** [reply]
created a ticket: https://github.com/JanDeDobbeleer/oh-my-posh/issues/5572
